### PR TITLE
Add `cosign verify-manifest` command

### PR DESCRIPTION
**Pull request by joshes**
Adds support for verifying kubernetes resources (manifests) and addresses #437 

It's a naive approach, but wanted to get feedback early on, especially around YAML parsing via regex and JSON support as discussed in the ticket.

### CODE REVIEW

File: cmd/cosign/cli/verify_manifest.go
Comment (hectorj2f) [inline cmd/cosign/cli/verify_manifest.go:49]: I believe this is coming from the verify-dockerfile command description.

File: cmd/cosign/cli/verify_manifest.go
Comment (hectorj2f) [inline cmd/cosign/cli/verify_manifest.go:116]: suggestion: Perhaps it could be more robust to decode the manifest and validate that the extensions are among those identified in the issue (Pod, DaemonSet...), then use the decoded object to get the image value for all the containers (e.g. `obj.Spec.Containers[*].Image`). 
Otherwise you could have a some resources that might not be among the identified list, e.g. CRs.

File: cmd/cosign/cli/verify_manifest.go
Comment (hectorj2f) [inline cmd/cosign/cli/verify_manifest.go:92]: suggestion: try to decode the yaml file to verify it is a valid.

File: cmd/cosign/cli/verify_manifest.go
Comment (hectorj2f) [reply cmd/cosign/cli/verify_manifest.go:116]: Let me know if you want me to help you with this.

Comment (joshes) [reply cmd/cosign/cli/verify_manifest.go:116]: @hectorj2f yes, this was a consideration as mentioned in https://github.com/sigstore/cosign/issues/437#issuecomment-887084109 - I think it would be better, but as different Kinds come out, it binds us more to the K8s API for parsing unless we still just walk down the tree looking for "image" keys (but guaranteed valid YAML) which would address your other comment.

This regex also doesn't account for quoted string variants, so either way this naive approach needs to change in _some_ fashion.

File: cmd/cosign/cli/verify_manifest.go
Comment (joshes) [reply cmd/cosign/cli/verify_manifest.go]: Updated the help docs to reflect.

File: cmd/cosign/cli/verify_manifest.go
Comment (dekkagaijin) [reply cmd/cosign/cli/verify_manifest.go:111]: Agreed. There's prior art for ducktyping `PodSpec`-like Objects, but I don't think it's worth blocking merge for a "perfect solution

File: cmd/cosign/cli/verify_manifest.go
Comment (dekkagaijin) [reply cmd/cosign/cli/verify_manifest.go:90]: (YAML or JSON)

Eventually we should probably just use a canonical k8s parser

**Review by dekkagaijin — APPROVED** [review]
I'm OK squeaking this in for 1.0.0, with the mentioned caveats...

File: cmd/cosign/cli/verify_manifest.go
Comment (hectorj2f) [reply cmd/cosign/cli/verify_manifest.go:111]: > There's prior art for ducktyping PodSpec-like Objects. 

Yes, you could use runtime.Raw and decode the object to the list of supported types here.

Sure, the current implementation should not be blocked by this. Once this PR is merged, I could open a PR with my proposed changes, so you can have a look at them.

**Comment by hectorj2f** [comment]
@joshes You need to sign off your commits.

**Comment by joshes** [comment]
> @joshes You need to sign off your commits.

Signed and pushed, thanks.

**Comment by dlorenc** [comment]
Amazing work! Let's get this in and follow up :)

**Pull request merged**
